File: corpus/04_binsearch.py
def bs(xs, t):
    lo, hi = 0, len(xs) - 1
    while lo <= hi:
        mid = (lo + hi) // 2
        if xs[mid] == t: return mid
        if xs[mid] < t: lo = mid + 1
        else: hi = mid - 1
    return -1
print(bs([1,3,5,7,9,11], 7))

File: corpus/04_binsearch.rs
fn bs(xs: &[i64], t: i64) -> i64 {
    let (mut lo, mut hi) = (0i64, xs.len() as i64 - 1);
    while lo <= hi {
        let mid = (lo + hi) / 2;
        if xs[mid as usize] == t {
            return mid;
        }
        if xs[mid as usize] < t {
            lo = mid + 1;
        } else {
            hi = mid - 1;
        }
    }
    -1
}

fn main() {
    println!("{}", bs(&[1, 3, 5, 7, 9, 11], 7));
}
